helix-term/build.rs

```diff
@@ -0,0 +1,11 @@
+use std::process::Command;
+fn main() {
+    let git_hash = Command::new("git")
+        .args(&["describe", "--dirty"])
```

Comment:
I think we also want to use `"rev-parse", "--short", "HEAD"` so we only get the SHA

Reply:
Oh, I thought @pickfire 's comment here:

https://github.com/helix-editor/helix/issues/922#issuecomment-955117911

made sense.  It seems like `dirty`-ness and distance from release could be really useful info.    Is there a constraint somewhere else that makes such a string difficult to work with?

Reply:
I had that for my first draft, and then changed to `describe --dirty` after reading @pickfire's comment on the issue. Would be no problem to switch back, if we're not concerned about version including knowledge of dirty state.

Reply:
I guess it's okay, but `describe` always uses the latest available tag. We'll have to be careful because if the code is built after a new tag is accidentally added then the version info will be wrong.

Reply:
Would it be worthwhile to use a specific process/tool to enforce that the version can't use any tag except the actual release tag?

Reply:
So this actually happened in the latest release (it builds as v0.5 on latest master because the release points to a commit before I rebased against master). I'd still rather see we _only_ pull the SHA and use the crate version.